test/functional/net.py

```diff
@@ -49,6 +82,19 @@ def run_test(self):
         assert_equal(self.nodes[0].getnetworkinfo()['networkactive'], True)
         assert_equal(self.nodes[0].getnetworkinfo()['connections'], 2)
 
+    def _test_getaddednodeinfo(self):
+        assert_equal(self.nodes[0].getaddednodeinfo(), [])
+        # add a node (node2) to node0
+        ip_port = "127.0.0.1:{}".format(p2p_port(2))
+        self.nodes[0].addnode(ip_port, 'add')
+        # check that the node has indeed been added
+        added_nodes = self.nodes[0].getaddednodeinfo(ip_port)
+        assert_equal(len(added_nodes), 1)
+        assert_equal(added_nodes[0]['addednode'], ip_port)
+        # check that a non-existant node returns an error
+        assert_raises_jsonrpc(-24, "Node has not been added",
+                              self.nodes[0].getaddednodeinfo, '1.1.1.1')
+
```

Comment:
seems like I added this function and line (ie upstream has this function and I added it in merging), any objections to keeping this function?

Reply:
you seem to have missed backporting bitcoin#10224, which contains more changes then this one. I'd suggest to backport that one first and put it before this backport.

Reply:
added